tests/Offer/CommandHandlers/AddLabelHandlerTest.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+namespace CultuurNet\UDB3\Offer\CommandHandlers;
+
+use Broadway\CommandHandling\Testing\CommandHandlerScenarioTestCase;
+use Broadway\EventHandling\EventBus;
+use Broadway\EventStore\EventStore;
+use CultuurNet\UDB3\Calendar;
+use CultuurNet\UDB3\CalendarType;
+use CultuurNet\UDB3\Event\EventRepository;
+use CultuurNet\UDB3\Event\Events\EventCreated;
+use CultuurNet\UDB3\Event\Events\LabelAdded;
+use CultuurNet\UDB3\Event\EventType;
+use CultuurNet\UDB3\Event\ValueObjects\LocationId;
+use CultuurNet\UDB3\Label;
+use CultuurNet\UDB3\Label\LabelServiceInterface;
+use CultuurNet\UDB3\Label\ReadModels\JSON\Repository\Entity;
+use CultuurNet\UDB3\Label\ReadModels\JSON\Repository\ReadRepositoryInterface;
+use CultuurNet\UDB3\Label\ValueObjects\LabelName;
+use CultuurNet\UDB3\Label\ValueObjects\Privacy;
+use CultuurNet\UDB3\Label\ValueObjects\Visibility;
+use CultuurNet\UDB3\Language;
+use CultuurNet\UDB3\Offer\Commands\AddLabel;
+use CultuurNet\UDB3\Offer\OfferRepository;
+use CultuurNet\UDB3\Place\PlaceRepository;
+use CultuurNet\UDB3\Title;
+use PHPUnit\Framework\MockObject\MockObject;
+use ValueObjects\Identity\UUID;
+use ValueObjects\StringLiteral\StringLiteral;
+
+final class AddLabelHandlerTest extends CommandHandlerScenarioTestCase
+{
+    /**
+     * @var ReadRepositoryInterface|MockObject
+     */
+    private $labelRepository;
+
+    /**
+     * @var Entity[]
+     */
+    private $mockedLabelReadModels;
+
+    /**
+     * @var LabelServiceInterface|MockObject
+     */
+    private $labelService;
+
+    protected function createCommandHandler(EventStore $eventStore, EventBus $eventBus): AddLabelHandler
+    {
+        $this->labelRepository = $this->createMock(ReadRepositoryInterface::class);
+        $this->labelRepository
+            ->method('getByName')
+            ->willReturnCallback(
+                function (StringLiteral $name) {
+                    return $this->mockedLabelReadModels[$name->toNative()] ?? null;
+                }
+            );
+
+        $this->labelService = $this->createMock(LabelServiceInterface::class);
+
+        return new AddLabelHandler(
+            new OfferRepository(
+                new EventRepository($eventStore, $eventBus),
+                new PlaceRepository($eventStore, $eventBus)
+            ),
+            $this->labelService,
+            $this->labelRepository
+        );
+    }
+
+    /**
+     * @test
+     */
+    public function it_should_use_existing_visibility_for_existing_labels(): void
```

Comment:
Do we need a bit more context in the name about "label adding" or that is implied from the handler?

Reply:
I figured it would be clear enough since the handler only has one responsibility